### PR TITLE
[MM-29945] docs: add config documentation

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Add doc for the configuration.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-29945

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+## ServiceSettings
+
+ServiceSettings is the configuration related to the web server.
+
+### Host
+
+*string*
+
+The URL that a Mattermost instance will use as S3 endpoint. The port number is required.
```
Comment (agnivade) [inline docs/config.md:11]: Technically it's not a URL, just a hostname with port.

File: docs/config.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+### HealthHost
+
+*string*
+
+The URL that will be used by K8s so that it can ping this endpoint to know the state of the server.
```
Comment (agnivade) [inline docs/config.md:17]: Same as above.

File: docs/config.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+### ResponseHeaderTimeout
+
+*int*
+
+Specifies the amount of time to wait for a server's response headers after fully writing the request.
+
+## S3Settings
+
+Settings related to S3-compatible object storage instance. E.g. Amazon S3, Minio etc.
```
Comment (agnivade) [inline docs/config.md:45]: I'd just avoid Minio for now. That's why the config struct is called `S3Settings`. Bifrost is just focused on S3 for now. It's good that minio is compatible with S3, but we don't guarantee that.

**Review by agnivade — APPROVED** [review]

File: docs/config.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+*string*
+
+The hostname and port that a Mattermost instance will use as S3 endpoint. The port number is required.
+
+### HealthHost
+
+*string*
+
+The hostname and port that will be used by K8s so that it can ping this endpoint to know the state of the server.
```
Comment (gabrieljackson) [inline docs/config.md:17]: Awesome!

**Review by gabrieljackson — APPROVED** [review]

**Pull request merged**